.travis.yml

```diff
@@ -1,36 +1,46 @@
 dist: xenial
```

Comment:
Changes on this file were introduced by me, basically to let Travis test on Ubuntu versions 16.04 (as before) and on 18.04 (new). The later ships with a more recent g++ version which would actually raise the warnings @sgtkellox was working on. They were not thrown with 16.04, i.e. we couldn't make out the different of @sgtkellox PR.